src/main/java/org/apache/cassandra/sidecar/tasks/HealthCheckPeriodicTask.java

```diff
@@ -101,6 +108,11 @@ public void execute(Promise<Void> promise)
 
         // join always waits until all its futures are completed and will not fail as soon as one of the future fails
         Future.join(futures)
+              .onComplete(v -> {
+                  int instancesUp = instancesConfig.instances().size() - instanceDown.get();
```

Comment:
you don't really need an AtomicInteger to keep track of the down instances. You can check the results instead.

`v.result().causes().stream().filter(Objects::nonNull).count()`

Reply:
But what if the futures failed due to some other reason, other than what is caught in `catch` block?

Reply:
how is that possible? we explicitly set the promise to either failed or succeeded

Reply:
`AtomicInteger instanceDown` provides better readability w/o perf sacrifice. I am leaning to the current implementation

Reply:
Makes sense. I tried these changes, we have to process the value of result to get cause. Maintaining the count outside seems simpler, wdyt?